diff -u mpich-1.2.0/MPI-2-C++/configure mpich-1.2.0/MPI-2-C++/configure
--- mpich-1.2.0/MPI-2-C++/configure	Tue Nov  9 23:24:53 1999
+++ mpich-1.2.0/MPI-2-C++/configure	Wed Jul 24 08:35:15 2002
@@ -1936,13 +1936,7 @@
   have_excep=no
 fi
 rm -f conftest*
-if test "$GXX" = "yes"; then
-    cat >> confdefs.h <<\EOF
-#define _MPIPP_USEEXCEPTIONS_ 0
-EOF
-
-    echo "$ac_t""no (g++ exceptions broken)" 1>&6
-elif test "$have_excep" = "yes"; then
+if test "$have_excep" = "yes"; then
     cat >> confdefs.h <<\EOF
 #define _MPIPP_USEEXCEPTIONS_ 1
 EOF
diff -u mpich-1.2.0/MPI-2-C++/configure.in mpich-1.2.0/MPI-2-C++/configure.in
--- mpich-1.2.0/MPI-2-C++/configure.in	Tue Nov  9 23:24:54 1999
+++ mpich-1.2.0/MPI-2-C++/configure.in	Wed Jul 24 08:31:32 2002
@@ -780,10 +780,7 @@
 
 AC_MSG_CHECKING([for throw/catch])
 AC_TRY_COMPILE(, int i=1; throw(i);, have_excep=yes, have_excep=no)
-if test "$GXX" = "yes"; then
-    AC_DEFINE(_MPIPP_USEEXCEPTIONS_, 0)
-    AC_MSG_RESULT([no (g++ exceptions broken)])
-elif test "$have_excep" = "yes"; then
+if test "$have_excep" = "yes"; then
     AC_DEFINE(_MPIPP_USEEXCEPTIONS_, 1)
     AC_MSG_RESULT([yes])
 else
